bundles/config/org.eclipse.smarthome.config.discovery.usbserial.linuxsysfs/src/main/java/org/eclipse/smarthome/config/discovery/usbserial/linuxsysfs/internal/PollingUsbSerialScanner.java

```diff
@@ -0,0 +1,177 @@
+        }
+    }
+
+    @Modified
+    protected synchronized void modified(Map<String, Object> config) {
+        if (config.containsKey(PAUSE_BETWEEN_SCANS_IN_SECONDS_ATTRIBUTE)) {
+            pauseBetweenScans = Duration
+                    .ofSeconds(parseLong(config.get(PAUSE_BETWEEN_SCANS_IN_SECONDS_ATTRIBUTE).toString()));
+
+            if (backgroundScanningJob != null) {
+                stopBackgroundScanning();
+                startBackgroundScanning();
+            }
+        }
+    }
+
+    /**
+     * Performs a single scan for newly added and removed devices.
+     */
+    @Override
+    public void doSingleScan() {
+        singleScanInternal(true);
+    }
+
+    /**
+     * Starts repeatedly scanning for newly added and removed USB devices in the usbserial devices folder (where the
+     * duration between two subsequent scans is configurable).
+     * <p/>
+     * This repeated scanning can be stopped using {@link #stopBackgroundScanning()}.
+     */
+    @Override
+    public synchronized void startBackgroundScanning() {
+        if (backgroundScanningJob == null) {
+            backgroundScanningJob = scheduler.scheduleWithFixedDelay(() -> {
+                singleScanInternal(false);
+            }, 0, pauseBetweenScans.getSeconds(), TimeUnit.SECONDS);
+            logger.debug("Scheduled usb-serial background discovery every {} seconds", pauseBetweenScans.getSeconds());
+        }
+    }
+
+    /**
+     * Stops repeatedly scanning for newly added and removed USB devices. This can be restarted using
+     * {@link #startBackgroundScanning()}.
+     */
+    @Override
+    public synchronized void stopBackgroundScanning() {
+        logger.debug("Stopping usb-serial background discovery");
+        ScheduledFuture<?> currentBackgroundScanningJob = backgroundScanningJob;
+        if (currentBackgroundScanningJob != null && !currentBackgroundScanningJob.isCancelled()) {
+            if (currentBackgroundScanningJob.cancel(true)) {
+                backgroundScanningJob = null;
+                logger.debug("Stopped usb-serial background discovery");
+            }
+        }
+    }
+
+    @Override
+    public void registerDiscoveryListener(UsbSerialDiscoveryListener listener) {
+        discoveryListeners.add(listener);
+    }
+
+    @Override
+    public void unregisterDiscoveryListener(UsbSerialDiscoveryListener listener) {
+        discoveryListeners.remove(listener);
+    }
+
+    private void singleScanInternal(boolean announceUnchangedDevices) {
+        try {
+            Delta<UsbSerialDeviceInformation> delta = deltaUsbSerialScanner.scan();
+            announceAddedDevices(delta.getAdded());
+            announceRemovedDevices(delta.getRemoved());
+            if (announceUnchangedDevices) {
+                announceAddedDevices(delta.getUnchanged());
+            }
+        } catch (IOException e) {
+            logger.warn("An IOException prevented a scan for USB serial devices: {}, message {}", e.getClass(),
```

Comment:
How about injecting the name at the proper place and using the simple-name of the exception class, i.e.
```java 
...warn("An {} prevented a scan for USB serial devices: {}", e.getClass().getSimpleName(), e.getMessage());
```

Reply:
Sounds good, as the log message will be more readable with your suggested change (except that "A" would be needed instead of "An" if a subclass of `IOException` does not start with a vowel, but I think that we can ignore this here...).